### PR TITLE
Allow notifications on open room if terminal not focused

### PR DESCRIPTION
**Pull request by Andrew-Collins**
Hi,

This PR is to implement #276.

It does this by putting the `focused` state of the application into the store and then checks it in the notifications handler.

This results in notifications being muted only if the room is currently displayed and the terminal is in focus.

If you would like a config entry to control this just let me know. :)

### CODE REVIEW

**Comment by benjajaja** [comment]
Looks great, thank you!

File: src/main.rs [hunk 1]
```diff
@@ -355,9 +355,13 @@ impl Application {
                     // Do nothing for now.
                 },
                 Event::FocusGained => {
+                    let mut store = self.store.lock().await;
+                    store.application.focused = true;
```
Comment (benjajaja) [inline src/main.rs:359]: Do we always get this event on startup?

Comment (Andrew-Collins) [reply src/main.rs:359]: Sorry busy week.
Thanks for the catch @benjajaja, I tested and we do not get the event on startup.
This has been resolved by setting the default of `focused` in the store to `true`.

Comment (benjajaja) [reply src/main.rs:359]: Thank you! There is an edge where iamb is started without focus, but I don't think it's possible to query the focus state from the terminal, and it's not a big deal.

**Review by benjajaja — APPROVED** [review]

File: src/main.rs (hunk 1 above)
Comment (Andrew-Collins) [reply src/main.rs:359]: Np :)

@benjajaja, I was also thinking about that edge case this morning, but I wasn't able to think of an easy way to resolve it either...

The only feasible way I can think of is to execute `crossterm` on entry to main, and run login/sync in the background.

That would end up being a fairly significant change, but if you are interested I can try and tackle it in another PR?

Comment (benjajaja) [reply src/main.rs:359]: I don't understand how that would give us the focus-state of the terminal. But I'm not really worried about that edge case, this PR is much more important. Thanks again!

**Comment by ulyssa** [comment]
@Andrew-Collins This is a great addition, thank you! :pray:

And I believe that @benjajaja is correct that we don't get the focused state on startup, but only whenever the window changes. (And that doesn't always happen with every terminal either). Defaulting `focused` to `true` is the right move. :+1:

**Pull request merged**